airflow_provider_azure_machinelearning/operators/machine_learning/endpoint.py

```diff
@@ -201,3 +201,84 @@ def execute(self, context: Context) -> None:
             raise KeyError(f"Failed to deploy endpoint {self.deployment.endpoint_name}.")
 
         return self.deployment.endpoint_name
+
+
+class AzureMachineLearningInvokeEndpointOperator(BaseOperator):
+    """
+    Executes Azure ML SDK to invoke an endpoint.
+
+    :param conn_id: The connection identifier for connecting to Azure Machine Learning.
+    :param endpoint: The spec of Azure ML endpoint
+    :waiting: When true, the operator blocks till the Azure ML endpoint deployment is completed.
+
+    Returns name(id) of the Azure ML endpoint.
+    """
+
+    ui_color = "white"
+    ui_fgcolor = "blue"
+
+    def __init__(
+        self,
+        *,
+        endpoint_name: str,
+        endpoint_type: str,
+        inputs: dict = {},
+        waiting: bool = False,
+        conn_id: str = None,
+        **kwargs,
+    ) -> None:
+
+        super().__init__(**kwargs)
+
+        self.endpoint_name = endpoint_name
+        self.endpoint_type = endpoint_type
+        self.inputs = inputs
+        self.waiting = waiting
+        self.conn_id = conn_id
+        self.hook = None
+        self.ml_client = None
+        self.job = None
+
+    def execute(self, context: Context) -> None:
+
+        self.log.info(
+            f"Executing { __class__.__name__} to deploy Endpoint {self.endpoint_name}."
+        )
+        self.hook = AzureMachineLearningHook(self.conn_id)
+        self.ml_client = self.hook.get_client()
+        
+        try:
+            self.get_caller().get(name=self.endpoint_name)
+        except AmlExceptions.ResourceNotFoundError:
+            self.log.info(f"Endpoint {self.endpoint_name} does not exist. Returning.")
+            return self.endpoint_name
+
+        self.log.info(f"Invoking Endpoint: {self.endpoint_name}.")
+        self.job = self.get_caller().invoke(endpoint_name=self.endpoint_name, inputs=self.inputs)
+
+        try:
+            if self.waiting:
+                self.ml_client.jobs.stream(name=self.job.name)
+                self.log.info(f"{self.job.name} has finished successfully.")
+        except Exception:
+            self.log.info(f"{self.job.name} has finished with errors.")
+            raise RuntimeError(f"{self.job.name} has finished with errors.")
+
+    def get_caller(self):
+        if self.endpoint_type == "online":
+            return self.ml_client.online_endpoints
+        elif self.endpoint_type == "batch":
+            return self.ml_client.batch_endpoints
+        else:
+            raise KeyError("Unrecognized Endpoint type.")
```

Comment:
If the only difference between online and batch operators is about the caller, I would recommend maybe explore a setup where you can have 3 classes

1. AzureMachineLearningInvokeBaseEndpointOperator (virtual base class, aka interface with undefined get_caller function that throws, but this class host rest oft the shared code from 2 classes below.)
2. AzureMachineLearningInvokeOnlineEndpointOperator (derived from 1st one and have a customer implementation of get_caller())
3. AzureMachineLearningInvokeOfflineEndpointOperator (derived from 1st one and have a customer implementation of get_caller())

What do you think>

Reply:
Hi saipulla,, 

I'm working on it, I'll check because I think you're right , online operator seems to be different to batch endpoint more than a get_caller_function. Naively,, I was trhinking that an online endpoint trigerred a job too, but it's not. 

So I used your implementation: 

AzureMachineLearningInvokeBaseEndpointOperator -> virtual class containing all common parameters and method 

AzureMachineLearningInvokeOnlineEndpointOperator -> class with some specific to online endpoint

AzureMachineLearningInvokeOnlineEndpointOperator -> class with some specific to batch endpoint

I'll make some tests and push the code, when it's good.

Reply:
Hi saipulla,

I tested AzureMachineLearningInvokeBaseEndpointOperator and there's a bug when Itried to invoke it in the same dag with the operator batch_deployment_task. Indeed in the dag example that's creating an batch endpoint, when airflow create a deployment it doesn't set it by default on attached endpoint. If I execute an invoke endpoint just after a deployment, following error is throw:  

Exception: No valid deployments to route to. Please check that the endpoint has at least one deployment with positive weight values or use a deployment specific header to route.

To work I think, we need to modify createDeploymentEndpoint by adding an argument to bind the new deployment on the endpoint by default. 

What do you think about it ?

Reply:
I think you can add optional parameter (default_deployment_name) to the operator to allow setting a default deployment.

BTW, there build pipelines below are showing some of your unit-tests are failing. Reminding you just in case you did not notice them already. 
![image](https://github.com/Azure/airflow-provider-azure-machinelearning/assets/28076103/7f75fed3-2e45-4f9d-ad6b-b28027150366)

Reply:
BTW, can you please include a sample DAG employing the new operator you introduced?
Maybe into this example dag file:
https://github.com/Azure/airflow-provider-azure-machinelearning/blob/main/airflow_provider_azure_machinelearning/example_dags/AML_Endpoint.py

Reply:
Yes i did this in the next commit. I just want to be sure that everything is ok for batch and online endpoint